docs/adr/0004-runtime-governance.md

```diff
@@ -0,0 +1,225 @@
+# ADR 0004: Runtime Governance
+
+## Changelog
+
+- 2020-09-17: Initial draft
+
+## Status
+
+Proposed
+
+## Context
+
+Currently all runtimes can only be governed by a single entity -- the runtime
+owner. In this regard governance means being able to update certain fields in
+the runtime descriptor stored by the consensus layer registry service. On one
+hand the runtime descriptor contains security-critical parameters and on the
+other there needs to be a mechanism through which the runtimes can be upgraded
+(especially so for TEE-based runtimes where a specific runtime binary is
+enforced via remote attestation mechanisms).
+
+This proposal extends runtime governance options and enables a path towards
+runtimes that can define their own governance mechanisms. This proposal assumes
+that [ADR 0003] has been adopted and runtimes can have their own accounts in the
+staking module.
+
+## Decision
+
+This proposal takes a simplistic but powerful approach which allows each runtime
+to choose its governance model upon its first registration. It does so through
+a newly introduced field in the runtime descriptor which indicates how the
+runtime descriptor can be updated in the future.
+
+Additionally it also introduces an additional safeguard for (all) runtime
+descriptor updates -- a configurable delay period after which any modifications
+to a runtime descriptor take effect instead of them taking effect immediately.
```

Comment:
what does this safeguard against?

Reply:
In theory it would give users time to e.g., move tokens out of the runtime in case they don't agree with the upgrade (security parameter change/MRENCLAVE change/etc.). Of course there are ways to prevent this, so it may not be worth adding the complexity.